src/pipeline.rs

```diff
@@ -60,6 +60,23 @@ impl Sink for Pipeline {
         debug_assert!(self.enqueued.is_empty());
         self.inner.poll_complete()
     }
+
+    fn close(&mut self) -> Poll<(), Self::SinkError> {
+        self.poll_complete()?;
```

Comment:
Does this change alone fix it by chance?

Reply:
I had the same thought before; it did not.